proposals/NNNN-widget-support.md

```diff
@@ -0,0 +1,248 @@
+</function>
+
+<function name="DeleteScreen" messagetype="request" since="5.1">
+  <description>
+    Deletes previously created screen of the SDL application.
+  </description>
+  <param name="screenID" type="Integer" mandatory="true" />
+</function>
+
+<function name="DeleteScreen" messagetype="response" since="5.1">
+  <!-- common response params -->
+</function>
+
+<enum "ScreenType">
+<element name="MAIN">
+  <description>
+    This Screen type describes the main screen on a display.
+  </description>
+</element>
+<element name="WIDGET">
+  <description>
+    A widget is a small screen that the app can create to provide information and softbuttons for a quick app control.
+  </description>
+</element>
+</enum>
+
+<enum "PredefinedScreens" since="5.1">
+<element name="DEFAULT_SCREEN" value="0">
+  <description>
+    The default screen is a main screen precreated on behalfs of the app.
+  </description>
+</element>
+</enum>
+```
+
+#### HMI API
+
+The HMI API should contain:
+- a copy of the enums `ScreenType` and `PredefinedScreens` in the "Common" interface
+- a copy of the functions `CreateScreen` and `DeleteScreen` in the "UI" interface
+
+#### Create & Delete screen
+
+The RPC `CreateScreen` allows an app to create a new screen on the display. The app needs to specify a screen ID that is used for screen manipulation e.g. with the RPC `Show` and the screen type which can either `MAIN` or `WIDGET` (see sub-section *Screen types*). 
+
+If desired the apps can duplicate content of an existing screen to the created screen using parameter `duplicateScreenID`. All RPCs sent to the screen with the ID equals `dulicateScreenID` will be duplicated for the creating screen. Bidirectional screen content duplication should not be supported. RPCs sent to the creating screen will not be duplicated to the screen with ID equals `duplicateScreenID`. 
+
+After a screen is successfully created the response will contain information and capabilities about the created screen.
+
+#### Screen types
+
+This proposal contains two types of screens.
+
+The main screen is the full size apps screen on a display. An app can have multiple main screens depending on the displays e.g. main screen on the central console and main screen on rear seat entertainment system. However it should not be allowed to have multiple main screen on a single display. Main screens follow HMI levels and permissions.
+
+The widget screen is a small screen type to provide quick information or shortcut softbuttons. Widgets don't follow HMI levels as main screen. Once a widget is created the all RPCs addressed to this widget follow policies based on HMI_LIMITED. This policy rule is due to the nature of widgets as they can be always avaiable to the user or should be ready to use when accessing the widget area (see alternative solution for widgets with HMI levels).
+
+Still widgets have affect to the HMI level of the app's main screen. Depending on the app policies:
```

Comment:
Change to:
Still widgets have **effects** to the HMI level of the app's main screen. Depending on the app policies: